src/main/java/org/apache/sling/pipes/internal/ThreadedPipe.java

```diff
@@ -0,0 +1,184 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.NonExistingResource;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.pipes.OutputWriter;
+import org.apache.sling.pipes.Pipe;
+import org.apache.sling.pipes.PipeBindings;
+import org.apache.sling.pipes.Plumber;
+import org.apache.sling.pipes.SuperPipe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This pipe executes the pipes it has in its configuration, in sequence or parallel;
+ * the output of the children pipes is merged;
+ * if execution is parallel, merge ordering is random;
+ * duplicate resources are kept in the output
+ */
+public class ThreadedPipe extends SuperPipe {
+    private static final Logger log = LoggerFactory.getLogger(ThreadedPipe.class);
+
+    public static final String RESOURCE_TYPE = "slingPipes/executor";
+    private static final String PN_QUEUE_SIZE = "queueSize";
+    private static final String PN_NUM_THREADS = "numThreads";
+
+    private int numThreads;
+    private ArrayBlockingQueue<Resource> outputQueue;
+
+    /**
+     * Constructor
+     * @param plumber plumber
+     * @param resource container's configuration resource
+     * @param upperBindings pipe bindings
+     * @throws Exception bad configuration handling
+     */
+    public ThreadedPipe(Plumber plumber, Resource resource, PipeBindings upperBindings) throws Exception{
+        super(plumber, resource, upperBindings);
+        int queueSize = properties.get(PN_QUEUE_SIZE, 10000);
+        numThreads = properties.get(PN_NUM_THREADS, 5);
+        outputQueue = new ArrayBlockingQueue<>(queueSize);
+    }
+
+    @Override
+    public void buildChildren() throws Exception {
+        for (Iterator<Resource> childPipeResources = getConfiguration().listChildren(); childPipeResources.hasNext();){
+            Resource pipeResource = childPipeResources.next();
+            Pipe pipe = plumber.getPipe(pipeResource, bindings);
+            if (pipe == null) {
+                log.error("configured pipe {} is either not registered, or not computable by the plumber", pipeResource.getPath());
+            } else {
+                pipe.setParent(this.getParent());
+                subpipes.add(pipe);
+            }
+        }
+    }
+
+    @Override
+    protected Iterator<Resource> computeSubpipesOutput() throws Exception {
+        return new ConcurrentIterator(numThreads);
+    }
+
+    private class PipeThread implements Runnable {
+
+        Pipe pipe;
+
+        PipeThread(Pipe pipe) {
+            this.pipe = pipe;
+        }
+
+        @Override
+        public void run() {
+            try {
+                plumber.execute(pipe.getResource().getResourceResolver().clone(null), pipe, null, new ThreadOutputWriter(), true);
```

Comment:
As discussed offline, we should support APIs that would work with AEM 6.3+ 
- either add plumber ```executeFork``` 
- or ```cloneResolver```  
api